Sources/SwiftParser/StringLiterals.swift

```diff
@@ -44,6 +148,243 @@ extension Parser {
     }
   }
 
+  /// If `text` only consists of indentation whitespace (space and tab), return
+  /// the trivia pieces that represent `text`, otherwise return `nil`.
+  private func parseIndentationTrivia(text: SyntaxText) -> [RawTriviaPiece]? {
+    let trivia = TriviaParser.parseTrivia(text, position: .leading)
+    if trivia.allSatisfy({ $0.isIndentationWhitespace }) {
+      return trivia
+    } else {
+      return nil
+    }
+  }
+
+  private func postProcessMultilineStringLiteral(
+    openQuote: RawTokenSyntax,
+    segments allSegments: [RawStringLiteralSegmentsSyntax.Element],
+    closeQuote: RawTokenSyntax
+  ) -> (
+    unexpectedBeforeOpenQuote: [RawTokenSyntax],
+    openQuote: RawTokenSyntax,
+    segments: [RawStringLiteralSegmentsSyntax.Element],
+    unexpectedBeforeCloseQuote: [RawTokenSyntax],
+    closeQuote: RawTokenSyntax
+  ) {
+    // -------------------------------------------------------------------------
+    // Precondition
+
+    assert(openQuote.trailingTriviaByteLength == 0, "Open quote produced by the lexer should not have trailing trivia because we would drop it during post-processing")
+    assert(closeQuote.leadingTriviaByteLength == 0, "Closing quote produced by the lexer should not have leading trivia because we would drop it during post-processing")
+    assert(
+      allSegments.allSatisfy {
+        if case .stringSegment(let segment) = $0 {
+          return segment.unexpectedBeforeContent == nil
+            && segment.unexpectedAfterContent == nil
+            && segment.content.leadingTriviaByteLength == 0
+            && segment.content.trailingTriviaByteLength == 0
+        } else {
+          return true
+        }
+      },
+      "String segement produced by the lexer should not have unexpected text or trivia because we would drop it during post-processing"
```

Comment:
```suggestion
      "String segment produced by the lexer should not have unexpected text or trivia because we would drop it during post-processing"
```